src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+  private readonly rundowns: Map<string, Rundown> = new Map()
+
+  private readonly rundownTimingContextSubjects: Map<string, BehaviorSubject<RundownTimingContext>> = new Map()
+
+  constructor(
+    private readonly rundownStateService: RundownStateService,
+    private readonly rundownTimingService: RundownTimingService
+  ) {
+    this.startLowResolutionTimer()
+  }
+
+  public async subscribeToRundownTimingContext(rundownId: string): Promise<Observable<RundownTimingContext>> {
+    const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> = await this.createRundownTimingContextSubject(rundownId)
+    return rundownTimingContextSubject.asObservable()
+  }
+
+  private startHighResolutionTimer(): void {
+    this.stopTimeResolutionTimer()
+    this.isHighResolutionTimer = true
+    this.timeResolutionTimerId = setInterval(this.onTimeTick.bind(this), HIGH_RESOLUTION_INTERVAL_DURATION_IN_MS)
+  }
+
+  private onTimeTick(): void {
+    this.rundowns.forEach(rundown => {
+      const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.getRundownTimingContextSubject(rundown.id)
+      if (!rundownTimingContextSubject) {
+        return
+      }
+      const currentEpochTime: number = Date.now()
+      const previousRundownTimingContext: RundownTimingContext = rundownTimingContextSubject.value
+      const expectedStartEpochTimeForRundown: number = this.rundownTimingService.getExpectedStartEpochTimeForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationInMsForRundown,
+        currentEpochTime
+      )
+      const expectedEndEpochTimeForRundown: number = this.rundownTimingService.getExpectedEndEpochTimeForRundown(rundown, previousRundownTimingContext.expectedDurationInMsForRundown, currentEpochTime)
+      const playedDurationInMsForOnAirPart: number = this.rundownTimingService.getPlayedDurationInMsForOnAirPart(rundown, currentEpochTime)
+      const playedDurationInMsForOnAirSegment: number = this.rundownTimingService.getPlayedDurationInMsForOnAirSegment(rundown, currentEpochTime)
+      const durationInMsSpentInOnAirSegment: number = this.rundownTimingService.getDurationInMsSpentInOnAirSegment(rundown, currentEpochTime)
+      const remainingDurationInMs: number = this.rundownTimingService.getRemainingDurationInMsForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationsInMsForSegments,
+        playedDurationInMsForOnAirSegment
+      )
+      const rundownTimingContext: RundownTimingContext = {
+        ...rundownTimingContextSubject.value,
+        currentEpochTime,
+        expectedStartEpochTimeForRundown,
+        expectedEndEpochTimeForRundown,
+        playedDurationInMsForOnAirPart,
+        playedDurationInMsForOnAirSegment,
+        durationInMsSpentInOnAirSegment,
+        remainingDurationInMsForRundown: remainingDurationInMs,
+      }
+      rundownTimingContextSubject.next(rundownTimingContext)
+    })
```

Comment:
Most of the values returned from the different methods are put directly into a `RundownTimingContext`. Why can't `RundownTimingService` just have a `getRundownTimingContext()` method? The only value in the `RundownTimingContext` that is not parsed to `RundownTimingService` is `rundownTimingContextSubject.value`.

So you could have something similar to:
```
const rundownTimingContext: RundownTimingContext = {
  ...rundownTimingContextSubject.value,
  ...rundownTimingService.getRundownTimingContext(rundown, currentEpochTime, previousRundownTimingContext.expectedDurationInMsForRundown)
}
```

Reply:
The `onRundownChanged` method handles the "static" values, that aren't affected by time alone, where `onTimeTick` updates the values that are affected by the passing of time.

To make it easier to get an overview of it, I have split up the functionality on `onTimeTick` into a method that handles updating the time related properties of a rundown timing context and a method for adjusting the timer interval.